spark-jobs/src/main/java/joinsomething/JoinSomething.java

```diff
@@ -0,0 +1,78 @@
+package joinsomething;
+
+import models.PageClick;
+import models.Person;
+import org.apache.spark.sql.*;
+
+import java.util.Properties;
+
+public class JoinSomething {
+    public static void main(String[] args) {
+        SparkSession sparkSession = SparkSession.builder()
+            .appName("joinSomething")
+            .getOrCreate();
+
+        String pageClicksPath = "/user/root/page-clicks/stream-*/part-*";
+        String peoplePath = "/user/root/peopleWithUUID.csv";
+        String cookieEmailsPath = "/user/root/userCookieEmailMappings.csv";
+
+        Properties connectionProperties = new Properties();
+        connectionProperties.put("driver", "com.mysql.jdbc.Driver");
+        connectionProperties.put("user", "user");
+        connectionProperties.put("password", "user");
+
+        Encoder<PageClick> pageClickEncoder = Encoders.bean(PageClick.class);
+        if (args.length >= 3) {
+            pageClicksPath = args[0];
+            peoplePath = args[1];
+            cookieEmailsPath = args[2];
+        }
+        Dataset<PageClick> pageClickDataset = sparkSession.read()
+            .json(pageClicksPath)
+            .as(pageClickEncoder);
+        //        pageClickDataset.show(5);
+
+        Encoder<Person> personEncoder = Encoders.bean(Person.class);
+        Dataset<Person> personDataset = sparkSession.read()
+            .option("header", "true")
+            .csv(peoplePath)
+            .as(personEncoder);
+
+        Dataset<Row> cookieEmailsDataset = sparkSession.read()
+            .option("header", "true")
+            .csv(cookieEmailsPath);
+
+        Dataset<Row> pageClickJoinCookie = pageClickDataset
+            .join(cookieEmailsDataset, "cookie");
+        Dataset<Row> joinByEmail = pageClickJoinCookie
+            .join(personDataset, "email");
+        joinByEmail.show(5);
+
+        Dataset<Row> select = joinByEmail.select("gender", "birthDate").cache();
+
+        RelationalGroupedDataset gender = joinByEmail.groupBy("gender");
+        Dataset<Row> genderCount = gender.count()
+                .withColumn("timestamp", functions.current_timestamp());
+        genderCount.show(5);
+
+        RelationalGroupedDataset year = select
+                .withColumn("year", select.col("birthDate").substr(0, 4))
+                .groupBy("year");
+        Dataset<Row> birthDateCount = year.count().sort("count")
+                .withColumn("timestamp", functions.current_timestamp());
+        System.out.println("Top 5 by year:");
+        birthDateCount.show(5);
+
+        System.out.println("Total count of rows: " + select.count());
+        System.out.println("Sum by year total: ");
+        birthDateCount.drop("birthDate").agg(functions.sum("count")).show();
+
+        genderCount.write()
+                .jdbc("jdbc:mysql://172.18.64.79:3306",
+                        "bdata.gender", connectionProperties);
+
+        birthDateCount.write()
+                .jdbc("jdbc:mysql://172.18.64.79:3306",
```

Comment:
Extract it to the global variable.  And substitute with this variable here and in 71 line